src/common/hooks/useEndorsementChain/helpers.ts

```diff
@@ -0,0 +1,166 @@
+import { Provider } from "@ethersproject/abstract-provider";
+import { TitleEscrowFactory__factory, TradeTrustToken } from "@govtechsg/token-registry/dist/contracts";
+import { Signer } from "ethers";
+import { EndorsementChain, TransferBaseEvent, TransferEvent, TransferEventType } from "../../../types";
+
+export const fetchEventTime = async (blockNumber: number, provider: Provider): Promise<number> => {
+  const eventTimestamp = (await (await provider.getBlock(blockNumber)).timestamp) * 1000;
+  return eventTimestamp;
+};
+
+export const extractEscrowAddress = async (
+  tokenRegistry: TradeTrustToken,
+  tokenId: string,
+  signer: Provider | Signer
```

Comment:
may want to extract this [type](https://github.com/TradeTrust/tradetrust-website/blob/a40e7a452a28bdc32bbdeeabee3a51635448205a/src/common/contexts/provider.tsx#L42) out to reuse. can avoid `import { Provider } from "@ethersproject/abstract-provider";` as well